### PR TITLE
Update address generator to match the flow-go

### PR DESCRIPTION
**Pull request by ramtinms**
## Description
Address generator on flow-go supports more type of networks, updating the files here to match the one on the flow-go, 
If won't impact previously existed network names here just support more networks like benchnet. 
______

For contributor use:

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-go-sdk/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: address.go [hunk 1]
```diff
@@ -261,8 +263,14 @@ func (a *Address) IsValid(chain ChainID) bool {
 // invalid code-words in the [64,45] code
 // these constants are used to generate non-Flow-Mainnet addresses
 const (
-	invalidCodeTestnet  = uint64(0x6834ba37b3980209)
-	invalidCodeEmulator = uint64(0x1cb159857af02018)
+	// invalidCodeTestNetwork is the invalid codeword used for long-lived test networks.
```
Comment (SaveTheRbtz) [inline address.go:266]: Should we keep a cross-note to keep them in sync with `model/flow/address.go` in flow-go?

File: flow.go [hunk 2]
```diff
@@ -95,14 +95,32 @@ func HashToStateCommitment(hash []byte) StateCommitment {
...
+	Mainnet ChainID = "flow-mainnet"
 
-// Testnet is the chain ID for the testnet node chain.
-const Testnet ChainID = "flow-testnet"
+	// Long-lived test networks
 
-// Emulator is the chain ID for the emulated node chain.
-const Emulator ChainID = "flow-emulator"
+	// Testnet is the chain ID for the testnet chain.
+	Testnet ChainID = "flow-testnet"
```
Comment (SaveTheRbtz) [inline flow.go:105]: same here for `model/flow/chain.go`

**Review by SaveTheRbtz — APPROVED** [review]
duplication is a bit tricky here.  We may consider eventually factoring these out into a submodule in flow-go repo, so that we can reference it w/o creating a dependency loop.

File: address.go [hunk 3]
```diff
@@ -107,11 +107,13 @@ func chainCustomizer(chain ChainID) uint64 {
 	case Mainnet:
 		return 0
 	case Testnet:
-		return invalidCodeTestnet
-	case Emulator:
-		return invalidCodeEmulator
+		return invalidCodeTestNetwork
+	case Stagingnet:
```
Comment (sideninja) [inline address.go:111]: could we cover these cases in tests as well?

**Review by sideninja — APPROVED** [review]

**Pull request merged**